### PR TITLE
Add read/write fbi format

### PR DESCRIPTION
**Pull request by alavenant**
Code for read and write Fbi format. 
TerrasScan use fbi format internaly to manipulate points clouds.

### CODE REVIEW

File: io/FbiReader.cpp [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+        for (size_t i(0); i<hdr->FastCnt; i++)
+        {
+            BYTE classif;
+            ifFbi->read(reinterpret_cast<char *>(&classif), hdr->BitsClass/8);
+            view->setField(Dimension::Id::Classification, i , int(classif));
+        }
+    }
+        
+    /*
+     Other Fbi contents - Not found equivalent in pdal model for now
```
Comment (hobu) [inline io/FbiReader.cpp:340]: It would be reasonable to make add PDAL dimensions for these.

File: io/FbiReader.cpp [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+using namespace pdal::fbi;
+
+namespace pdal
+{
+
+static StaticPluginInfo const s_info
+{
+    "readers.fbi",
+    "Bbi Reader",
+    "http://pdal.io/stages/readers.fbi.html", // _ToDo_
```
Comment (hobu) [inline io/FbiReader.cpp:51]: We will need a doc to merge this.

File: io/FbiWriter.cpp [hunk 3]
```diff
@@ -0,0 +1,424 @@
...
+using namespace pdal;
+using namespace pdal::fbi;
+
+namespace
+{
+    const StaticPluginInfo s_info
+    {
+        "writers.fbi",
+        "FBI Writer",
+        "http://pdal.io/stages/writers.fbi.html"
```
Comment (hobu) [inline io/FbiWriter.cpp:46]: need a doc

File: test/unit/io/FbiReaderTest.cpp [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+namespace pdal
+{
+
+
+namespace
+{
+
+std::string getTestfilePath()
+{
+    return Support::datapath("fbi/test.fbi");
```
Comment (hobu) [inline test/unit/io/FbiReaderTest.cpp:50]: Can you write the `1.2-with-color.las` file to fbi, add it to the test suite, and add a simple test that checks coordinates or intensity or something? This provides commonly tested values for various formats so we have some context when looking at transformed data.

**Review by hobu — CHANGES_REQUESTED** [review]
This is great! Thanks for the contribution! A few small comments, it will need some docs, and a small tweak to the test config will help it fit with other format drivers. We will plan to include this in PDAL 2.4 if it can be cleaned up in the next few weeks.

**Review by abellgithub — COMMENTED** [review]
General:

All this seems to assume you're running on a little-endian machine. This is pretty-much always the case these days, but if you're not going to support big-endian, you should probably throw an error if someone tries to use it on BE.

**Comment by hobu** [comment]
Moved over to #3822 so I can push and finish this up. Thanks for the contribution!

**Pull request closed without merging**